bin/setup.csh

```diff
@@ -0,0 +1,46 @@
+# setup lsstsw environment
+#
+# source this file from your ~/.cshrc
+#
+# possibly blank, but can't be parsed as $_
+set cmd=($_)
+if ( $?0 ) then
+    set source=$0
+else if ( $#cmd >= 3 ) then
+    set source=`pwd`/${cmd[2]}
+else if (-f /usr/sbin/lsof ) then
+    set source=`/usr/sbin/lsof +p $$ | grep -oE /.\*setup.csh`
```

Comment:
I really hadn't expected people to be adding these to their `.login` file but if you have to do something more complicated to support that working then please ignore my comments. Maybe add a comment to the file though to explain the use case.